tests/test_pickling.py

```diff
@@ -9,18 +9,41 @@
 from pybind11_tests import pickling as m
 
 
-def test_pickle_simple_callable():
+def test_assumptions():
+    assert pickle.HIGHEST_PROTOCOL >= 0
+
+
+@pytest.mark.parametrize("protocol", range(pickle.HIGHEST_PROTOCOL + 1))
```

Comment:
Interesting, we don't normally support protocol 0 with `py::pickle`, but I guess this does?

Reply:
I changed it like this:

```diff
-def test_assumptions():
+def all_pickle_protocols():
     assert pickle.HIGHEST_PROTOCOL >= 0
+    return range(pickle.HIGHEST_PROTOCOL + 1)
 
 
-@pytest.mark.parametrize("protocol", range(pickle.HIGHEST_PROTOCOL + 1))
+@pytest.mark.parametrize("protocol", all_pickle_protocols())
 def test_pickle_simple_callable(protocol):
```

Basically, I'm paranoid about accidentally not running `test_pickle_simple_callable()` at all.

This is easily overlooked:
```
test_pickling.py::test_pickle_simple_callable[protocol0] SKIPPED (got empty parameter set ['protocol'], function test_pickle_simple_...)
```

Reply:
Oof, sorry, the above was meant to be a response to the other question.

> Interesting, we don't normally support protocol 0 with `py::pickle`, but I guess this does?

Yes, this is completely independent. There is no reason to limit what protocol versions are accepted.

Reply:
If pickle reports a maximum pickle version of 0, pickle would be broken. That would be a Python bug, I don't think we should be worried about Python bugs of that magnitude. Especially since it only would produce a minor warning in the test suite.